### PR TITLE
feat: Switch to esbuild (common, api, chart)

### PR DESCRIPTION
**Pull request by GordonSmith**
## Checklist:
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [ ] The change has been fully tested:
  - [ ] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

File: packages/common/src/PropertyExt.ts [hunk 1]
```diff
@@ -769,25 +769,25 @@ export class PropertyExt extends Class {
...
-
-export function publishProxy(proxy: string, method?: string, defaultValue?, ext: { reset?: boolean } = {}) {
-    return function (target: any, key: string) {
-        if (ext.reset) {
-            target.publishReset();
-        }
-        target.publishProxy(key, proxy, method, defaultValue);
-    };
-}
+// export function publish(defaultValue, type?: PublishTypes, description?: string, set?: string[] | (() => string[] | Array<{ value: string, text: string }>) | IPublishExt, ext: IPublishExt = {}) {
```
Comment (jeclrsg) [inline packages/common/src/PropertyExt.ts:772]: Just wanted to point these couple commented out functions; they seemed different from all the other classes w/ the commented out blocks of properties

File: packages/chart/tests/chart.browser.spec.ts [hunk 2]
```diff
@@ -33,7 +33,10 @@ describe("@hpcc-js/chart", () => {
         const item = (chart as any)[key];
         if (item) {
             if (!urlSearch || urlSearch === item.prototype.constructor.name) {
-                describe(`${item.prototype.constructor.name}`, () => {
+                describe(`${item.prototype?.constructor?.name}`, () => {
+                    it("Simple", () => {
+                        expect(true).to.be.true;
+                    });
```
Comment (jeclrsg) [inline packages/chart/tests/chart.browser.spec.ts:39]: I noticed a few of these being added (here and in some other spec files) & just didn't understand what they were testing

File: packages/api/tests/args.spec.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { describe, it, expect } from "vitest";
+
+describe("arguments", function () {
```
Comment (jeclrsg) [inline packages/api/tests/args.spec.ts:3]: I don't follow what this test file is for

**Review by jeclrsg — COMMENTED** [review]
@GordonSmith There was one file that has a couple commented out functions that seemed different than the other classes w/ the commented out properties. And just a couple questions about some of the tests

File: packages/chart/tests/chart.browser.spec.ts (hunk 2 above)
Comment (GordonSmith) [reply packages/chart/tests/chart.browser.spec.ts:39]: A `describe` with no `it` will cause an error, this ensures there is always at least 1 `it`

File: packages/api/tests/args.spec.ts (hunk 3 above)
Comment (GordonSmith) [reply packages/api/tests/args.spec.ts:3]: It was me sanity checking `arguments` and TypeScript strict checking, will remove.

**Review by jeclrsg — APPROVED** [review]

File: packages/common/src/PropertyExt.ts (hunk 1 above)
Comment (GordonSmith) [reply packages/common/src/PropertyExt.ts:772]: They were the decorator function `@publish` and `@publishProxy`, which are still not part of the standard.

**Comment by GordonSmith** [comment]
@jeclrsg I removed the commented code and stray test.

**Review by jeclrsg — APPROVED** [review]

**Pull request merged**